llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -21808,45 +21822,40 @@ SDValue tryLowerPartialReductionToDot(SDNode *N,
 SDValue tryLowerPartialReductionToWideAdd(SDNode *N,
                                           const AArch64Subtarget *Subtarget,
                                           SelectionDAG &DAG) {
-
-  assert(N->getOpcode() == ISD::INTRINSIC_WO_CHAIN &&
-         getIntrinsicID(N) ==
-             Intrinsic::experimental_vector_partial_reduce_add &&
-         "Expected a partial reduction node");
-
   if (!Subtarget->isSVEorStreamingSVEAvailable())
```

Comment:
This existing code looks wrong to me because the wide add instructions are only available in SVE2 (or SME in streaming mode).  I'm pretty sure this code will lead to an internal compiler error when only SVE is available and is worth fixing asap as a new PR.

Reply:
Done